### PR TITLE
Add native SSH workspace entrypoint

### PR DESCRIPTION
**Pull request by austinywang**
## Summary
- add a native File > New SSH Workspace action, command palette command, and customizable unbound shortcut
- reuse the existing managed cmux ssh launcher so native SSH workspaces keep the current remote routing, reconnect, persistence, and upload behavior
- add localized dialog/validation strings plus docs and README updates

## Scope
This is a first increment for #1664. It provides a first-class native entrypoint to the existing managed SSH workspace path. Saved connection profiles, zellij-oriented setup UX, and richer session management UI remain follow-up work.

## Validation
- jq empty Resources/Localizable.xcstrings
- git diff --check

Local builds/tests were not run per the issue workflow instructions; CI is the first compiler/test pass.

---
<a href="https://app.blacksmith.sh/manaflow-ai/codesmith/cmux/pr/4743"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with Codesmith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1782351328&installation_id=133855650&pr_number=4743&repository=manaflow-ai%2Fcmux&return_to=https%3A%2F%2Fgithub.com%2Fmanaflow-ai%2Fcmux%2Fpull%2F4743&signature=c573f8d51fad262420aeb1dcd1605706b80fc50a889458aa5fa79f1cae594171"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with Codesmith</picture></a>
<sup>Need help on this PR? Tag <code>@codesmith</code> with what you need. Autofix is disabled.</sup>

---

> [!NOTE]
> **Medium Risk**
> Touches SSH connection input validation and launches remote sessions with identity paths; mitigated by reuse of the existing launcher and strict manual parsing, but still security-adjacent.
> 
> **Overview**
> Adds a **native “New SSH Workspace”** flow so users can open managed remote workspaces from the app without typing `cmux ssh` on the command line.
> 
> **File** menu, the command palette, and an optional (default unbound) keyboard shortcut all call `performNewSSHWorkspaceAction`, which shows a connect dialog for destination, port, identity file, and workspace name. Input is validated (length, port range, hidden/unsafe characters, SSH config–style hosts, bracketed IPv6) and passed through new `CmuxSSHURLRequest.manual` logic. Successful connects reuse `CmuxSSHURLProcessLauncher` and extend CLI args with `--identity`, `--name`, and `--window` for the active window.
> 
> Docs/README note the menu entry and `shortcuts.bindings.newSSHWorkspace`; en/ja strings and unit tests cover manual requests and validation edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit ba82aa84449641e76f7683ce4e53c36f10cc77f8. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

---
## Summary by cubic
Adds a native “New SSH Workspace…” in the File menu and command palette (with an optional shortcut) to open managed SSH workspaces without the CLI. Reuses the existing SSH transport, targets the active window for correct routing, and forwards identity/name/window; first step toward #1664.

- **New Features**
  - Native connect dialog (destination, port, identity file, workspace name) with validation for length, hidden/unsafe chars, leading dashes, and port range; accepts SSH config pattern hosts and bracketed IPv6, rejects unbracketed IPv6.
  - Launch path reuses the managed SSH flow and builds CLI with --port/--identity/--name/--window for the active window; command palette action wired; localized strings (en/ja); README/docs updates; tests for manual request build, host patterns, IPv6 accept/reject, and unsafe inputs; clearer URL error for unsafe identity file paths.

- **Migration**
  - Optional: bind the shortcut via `shortcuts.bindings.newSSHWorkspace` (default unbound, e.g., "cmd+shift+s").

<sup>Written for commit ba82aa84449641e76f7683ce4e53c36f10cc77f8. Summary will update on new commits. <a href="https://cubic.dev/pr/manaflow-ai/cmux/pull/4743?utm_source=github">Review in cubic</a></sup>

## Summary by CodeRabbit

* **New Features**
  * Introduced "New SSH Workspace" feature accessible via File → New SSH Workspace… menu and command palette for creating SSH connections with configurable destination, identity file, and port settings.

* **Documentation**
  * Updated README and configuration guide with instructions for the new SSH workspace feature and information on customizing the keyboard shortcut binding.

[Review Change Stack](https://app.coderabbit.ai/change-stack/manaflow-ai/cmux/pull/4743?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: Sources/ContentView.swift [hunk 1]
```diff
@@ -7440,6 +7448,14 @@ struct ContentView: View {
                 debugSource: "palette.newWorkspace"
             )
         }
+        registry.register(commandId: "palette.newSSHWorkspace") {
+            DispatchQueue.main.async {
+                AppDelegate.shared?.performNewSSHWorkspaceAction(
+                    preferredWindow: NSApp.keyWindow ?? NSApp.mainWindow,
+                    debugSource: "palette.newSSHWorkspace"
+                )
+            }
```
[the remark this answers is not in the document]
Comment (austinywang) [reply Sources/ContentView.swift:7508]: Fixed by resolving the originating main window from windowId inside the command palette handler, instead of reading NSApp.keyWindow after the palette closes.

*— Claude Code*

File: Sources/AppDelegate+CmuxSSHURL.swift [hunk 2]
```diff
@@ -132,7 +132,225 @@ private final class CmuxSSHURLConfirmationGate: NSObject {
...
+            )
+        case .invalidPort:
+            return String(
+                localized: "dialog.newSSHWorkspace.validation.invalidPort",
+                defaultValue: "The SSH port must be between 1 and 65535."
+            )
+        default:
+            return cmuxSSHURLParseErrorMessage(error)
+        }
+    }
```
[the remark this answers is not in the document]
Comment (austinywang) [reply Sources/AppDelegate+CmuxSSHURL.swift]: Fixed by replacing the URL-message fallback with an explicit dialog-specific fallback for URL-only parse cases.

*— Claude Code*

File: Sources/AppDelegate+CmuxSSHURL.swift [hunk 3]
```diff
@@ -132,7 +132,225 @@ private final class CmuxSSHURLConfirmationGate: NSObject {
...
+    func performNewSSHWorkspaceAction(
+        preferredWindow: NSWindow? = nil,
+        debugSource: String = "newSSHWorkspace"
+    ) -> Bool {
+        var draft = CmuxNewSSHWorkspaceDialogDraft()
+        let targetWindow = preferredWindow ?? NSApp.keyWindow ?? NSApp.mainWindow
+        let targetWindowId = targetWindow.flatMap { mainWindowId(from: $0) }
+        while true {
+            guard let nextDraft = promptForNewSSHWorkspace(draft: draft, preferredWindow: preferredWindow) else {
+#if DEBUG
```
[the remark this answers is not in the document]
Comment (austinywang) [reply Sources/AppDelegate+CmuxSSHURL.swift]: Fixed by passing the resolved targetWindow into promptForNewSSHWorkspace so the dialog uses the same window selected for the SSH request.

*— Claude Code*

File: Sources/CmuxSSHURLRequest.swift [hunk 4]
```diff
@@ -356,4 +417,82 @@ struct CmuxSSHURLRequest: Equatable {
...
+                return .failure(.destinationStartsWithDash)
+            }
+            guard isAllowedSSHUser(user), isAllowedSSHHost(host) else {
+                return .failure(.destinationContainsUnsafeCharacters)
+            }
+        } else {
+            guard isAllowedSSHHost(value) else {
+                return .failure(.destinationContainsUnsafeCharacters)
+            }
+        }
```
[the remark this answers is not in the document]
Comment (austinywang) [reply Sources/CmuxSSHURLRequest.swift:464]: Fixed by widening manual host validation to allow SSH config-style host patterns while still rejecting whitespace, hidden characters, and ambiguous @ characters.

*— Claude Code*

File: Sources/AppDelegate+CmuxSSHURL.swift [hunk 5]
```diff
@@ -132,7 +132,225 @@ private final class CmuxSSHURLConfirmationGate: NSObject {
...
+        alert.accessoryView = fields.accessoryView()
+        alert.window.initialFirstResponder = fields.destinationField
+
+        let response: NSApplication.ModalResponse
+        if let preferredWindow {
+            preferredWindow.makeKey()
+            response = alert.runModal()
+        } else {
+            response = alert.runModal()
+        }
```
[the remark this answers is not in the document]
Comment (austinywang) [reply Sources/AppDelegate+CmuxSSHURL.swift]: Cleaned up the duplicated runModal branches; the preferred window is made key before the single modal call.

*— Claude Code*

File: Sources/CmuxSSHURLRequest.swift [hunk 6]
```diff
@@ -356,4 +417,82 @@ struct CmuxSSHURLRequest: Equatable {
...
+            return .failure(.invalidPort)
+        }
+        return .success(port)
+    }
+
+    private static func normalizedManualIdentityFile(_ rawValue: String?) -> String? {
+        guard let rawValue else { return nil }
+        let value = rawValue.trimmingCharacters(in: .whitespacesAndNewlines)
+        return value.isEmpty ? nil : value
+    }
```
[the remark this answers is not in the document]
Comment (austinywang) [reply Sources/CmuxSSHURLRequest.swift]: Fixed by validating identity file paths for hidden control/formatting characters and adding a regression test for that path.

*— Claude Code*

File: Sources/AppDelegate+CmuxSSHURL.swift [hunk 7]
```diff
@@ -132,7 +132,235 @@ private final class CmuxSSHURLConfirmationGate: NSObject {
...
+                return false
+            }
+            draft = nextDraft
+
+            switch CmuxSSHURLRequest.manual(
+                destination: draft.destination,
+                port: draft.port,
+                identityFile: draft.identityFile,
+                title: draft.title,
+                windowId: targetWindowId
```
[the remark this answers is not in the document]
Comment (austinywang) [reply Sources/AppDelegate+CmuxSSHURL.swift:261]: Fixed by letting performNewSSHWorkspaceAction accept a preferredTabManager and routing the File menu action through activeTabManager, matching the adjacent New Workspace command path.

*— Claude Code*

File: Sources/CmuxSSHURLRequest.swift [hunk 8]
```diff
@@ -356,4 +426,92 @@ struct CmuxSSHURLRequest: Equatable {
...
+                return .failure(.destinationStartsWithDash)
+            }
+            guard isAllowedSSHUser(user), isAllowedManualSSHHost(host) else {
+                return .failure(.destinationContainsUnsafeCharacters)
+            }
+        } else {
+            guard isAllowedManualSSHHost(value) else {
+                return .failure(.destinationContainsUnsafeCharacters)
+            }
+        }
```
[the remark this answers is not in the document]
Comment (austinywang) [reply Sources/CmuxSSHURLRequest.swift:464]: Fixed by routing colon-bearing manual hosts through the stricter SSH host validator. Bracketed IPv6 remains accepted, while unbracketed IPv6 forms such as alice@::1 and ::1 are rejected with behavior coverage.

*— Claude Code*
